src/components/Header/Header.js

```diff
@@ -125,26 +122,20 @@ class Header extends React.Component {
   }
 
   componentWillUnmount() {
-    HeaderStore.unlisten(this.onChange.bind(this));
     // Listen on FeatureFlags Store updates
     FeatureFlags.store.unlisten(this.onFeatureFlagsChange.bind(this));
     // Removing event listener to minimize garbage collection
   }
 
   onChange() {
-    this.setState(
-      _extend(
-        {
-          loginCookieValue: this.state.loginCookieValue,
-          patronName: this.state.patronName,
-          patronInitial: this.state.patronInitial,
-          patronDataReceived: this.state.patronDataReceived,
-          isFeatureFlagsActivated: {},
-          logOutUrl: this.state.logOutUrl,
-        },
-        HeaderStore.getState()
-      )
-    );
+    this.setState({
+      loginCookieValue: this.state.loginCookieValue,
+      patronName: this.state.patronName,
+      patronInitial: this.state.patronInitial,
+      patronDataReceived: this.state.patronDataReceived,
+      isFeatureFlagsActivated: {},
+      logOutUrl: this.state.logOutUrl,
+    });
```

Comment:
What is achieved by setting a state variable to the value of that variable in state? I think `setState` only sets those values that you pass in, so I don't think there's a need to pass every state variable in if you're not chanigng it. This seems like it should have no effect? Does this trigger something nonobvious?

Reply:
Right, in this case, it's not needed here. Not sure how it ended up here but will remove it.